x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/description_step/index.tsx

```diff
@@ -111,14 +110,14 @@ export const addFilterStateIfNotThere = (filters: Filter[]): Filter[] => {
 const getDescriptionItem = (
   field: string,
   label: string,
-  value: unknown,
+  data: unknown,
   filterManager: FilterManager,
   indexPatterns?: IIndexPattern
 ): ListItems[] => {
   if (field === 'queryBar') {
-    const filters = addFilterStateIfNotThere(get('queryBar.filters', value) ?? []);
-    const query = get('queryBar.query', value) as Query;
-    const savedId = get('queryBar.saved_id', value);
+    const filters = addFilterStateIfNotThere(get('queryBar.filters', data) ?? []);
+    const query = get('queryBar.query.query', data);
+    const savedId = get('queryBar.saved_id', data);
```

Comment:
All of this code prior to you is not type safe as there is an unknown but no "type narrowing" but rather just using a get. As you can probably tell you if you change a variable name you have to understand and know what that variable is in a bunch of strings which is incredibly prone to bugs.

I would strongly recommend as a team and individuals to stop using the `get` as much as possible in these unsafe manners.

For one, it returns any `any` and if it does you have to cast is where we don't always cast it and suddenly have an "any floating around:

<img width="380" alt="Screen Shot 2020-03-17 at 10 25 50 AM" src="https://user-images.githubusercontent.com/1151048/76877959-bd811780-6839-11ea-88ad-0cb78d7a88ec.png">

Second the casts previous to your code is making an assumption that everything will have a value, but the definition of lodash get is that it can _always_ return a `undefined` but since the types aren't correct with it, most developers ignore that.

Third as I refactor just this bit to use interfaces you will notice other assumptions this code accidentally makes and does not safe guard its self. So either the types are making assumptions that are not true (such as optional undefined) or this code existing before you had potential bugs.

A better alternative is to just do an unsafe cast right at the start of this function and get back type safety after the fact like so:

```ts
  if (field === 'queryBar') {
    // TODO: use io-ts or a TypeGuard "is" to correctly narrow this later.
    const queryBar = data as FieldValueQueryBar;
    const filters = addFilterStateIfNotThere(queryBar.filters ?? []);
    // TODO: TypeNarrow or handle the type that this might not be a string but some non-string. For now we assume this is a string
    const query = queryBar.query.query as string;
    const savedId = queryBar.saved_id ?? '';
    return buildQueryBarDescription({
      field,
      filters,
      filterManager,
      query,
      savedId,
      indexPatterns,
    });
```

That's the first part. For the rest below I would just do the same dangerous "as cast" as at least that would be better to get back types for when variable names change rather than hunting around for strings (which is really bad for maintenance)

Reply:
Deferred to https://github.com/elastic/kibana/issues/60567